pkg/query/querier.go

```diff
@@ -209,7 +209,11 @@ func (q *querier) Select(_ bool, hints *storage.SelectHints, ms ...*labels.Match
 	// The querier has a context but it gets canceled, as soon as query evaluation is completed, by the engine.
 	// We want to prevent this from happening for the async store API calls we make while preserving tracing context.
 	ctx := tracing.CopyTraceContext(context.Background(), q.ctx)
```

Comment:
Ok, so I know where is confusion from (👍🏽 
This exact line *on purpose* removes cancellation (plus timeout). Adding it below "maybe" makes sense, but we need to also think if 

> // We want to prevent this from happening for the async store API calls we make while preserving tracing context.

Is still what we want.... 🤔